api/v2/views/provider.py

```diff
@@ -1,27 +1,28 @@
-from rest_framework import viewsets
-from rest_framework.permissions import IsAuthenticated, IsAdminUser
 from rest_framework.decorators import detail_route
+
 from core.models import Provider, Group
+from core.query import only_current_provider
+
+from api.permissions import CloudAdminRequired
 from api.v2.serializers.details import ProviderSerializer
 from api.v2.serializers.summaries import SizeSummarySerializer
-from core.query import only_current_provider
+from api.v2.views.base import AuthReadOnlyViewSet, AuthViewSet
 
 
-class ProviderViewSet(viewsets.ReadOnlyModelViewSet):
+class ProviderViewSet(AuthReadOnlyViewSet):
     """
     API endpoint that allows providers to be viewed or edited.
     """
+
     queryset = Provider.objects.all()
     serializer_class = ProviderSerializer
-    permission_classes = (IsAuthenticated,)
     http_method_names = ['get', 'head', 'options', 'trace']
 
     def get_permissions(self):
         method = self.request.method
         if method == 'DELETE' or method == 'PUT':
-            self.permission_classes = (IsAdminUser,)
-
-        return super(viewsets.GenericViewSet, self).get_permissions()
+            self.permission_classes += (CloudAdminRequired,)
```

Comment:
Could be refactored similar to tags using self.request.method in ['DELETE', 'PUT]